test/e2e/live_iso_test.go

```diff
@@ -65,6 +83,36 @@ func liveIsoTest() {
 	})
 
 	AfterEach(func() {
-		dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, e2eConfig.GetIntervals, clusterName, clusterctlLogFolder, skipCleanup)
+		By("Deleting ISO bmh")
+
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+
+		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+		Expect(err).NotTo(HaveOccurred(), "Error getting BMHs")
+		bmhsLen := len(bmhs)
+
+		for _, bmh := range bmhs {
+			Logf("checking bmh %s", bmh.Name)
+			if bmh.Status.Provisioning.State == bmov1alpha1.StateProvisioned {
+				Logf("ISO bmh found %s", bmh.Name)
+				bmh.Spec.Online = true
+				bmh.Spec.Image = &bmov1alpha1.Image{
+					URL:          "",
+					Checksum:     "",
+					ChecksumType: "",
+					DiskFormat:   nil,
+				}
+				Expect(bootstrapClient.Update(ctx, &bmh)).NotTo(HaveOccurred())
+			}
+		}
+		By("Waiting for deprovisioned ISO bmh to be available")
+		Eventually(
+			func(g Gomega) {
+				bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+				Expect(err).NotTo(HaveOccurred(), "Error getting BMHs")
+				filtered := filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateAvailable)
+				g.Expect(len(filtered)).To(Equal(bmhsLen))
+			}, e2eConfig.GetIntervals(specName, "wait-bmh-available")...).Should(Succeed())
+		By("ISO bmh deleted successfully")
```

Comment:
is it a leftover, I do not find the step where we delete BMH?

Reply:
It should say deprovisioned successfully, I can change it.